addons/report/models/abstract_report.py

```diff
@@ -1,37 +1,33 @@
 # -*- coding: utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
-from openerp.osv import osv
+from odoo import api, models
 
 
-class AbstractReport(osv.AbstractModel):
+class AbstractReport(models.AbstractModel):
     """Model used to embed old style reports"""
     _name = 'report.abstract_report'
     _template = None
     _wrapped_report_class = None
 
-    def render_html(self, cr, uid, ids, data=None, context=None):
-        context = dict(context or {})
-
+    @api.multi
+    def render_html(self, data=None):
+        context = dict(self.env.context)
         # If the key 'landscape' is present in data['form'], passing it into the context
         if data and data.get('form', {}).get('landscape'):
             context['landscape'] = True
-
-        if context and context.get('active_ids'):
+        if context.get('active_ids'):
             # Browse the selected objects via their reference in context
             model = context.get('active_model') or context.get('model')
-            objects_model = self.pool[model]
-            objects = objects_model.browse(cr, uid, context['active_ids'], context=context)
+            objects = self.env[model].browse(context['active_ids'])
         else:
             # If no context is set (for instance, during test execution), build one
-            model = self.pool['report']._get_report_from_name(cr, uid, self._template).model
-            objects_model = self.pool[model]
-            objects = objects_model.browse(cr, uid, ids, context=context)
+            model = self.env['report']._get_report_from_name(self._template).model
+            objects = self.env[model].browse(self.ids)
             context['active_model'] = model
-            context['active_ids'] = ids
-
+            context['active_ids'] = self.ids
         # Generate the old style report
-        wrapped_report = self._wrapped_report_class(cr, uid, '',  context=context)
+        wrapped_report = self._wrapped_report_class(self.env.cr, self.env.uid, '', context=context)
```

Comment:
can you check with self._model._wrapped_report_class(self.env.cr, self.env.uid, '', context=context)
because self is recordset.